cpp/src/arrow/flight/sql/odbc/flight_sql/accessors/date_array_accessor_test.cc

```diff
@@ -38,7 +38,11 @@ using arrow::ArrayFromVector;
 using odbcabstraction::GetTimeForSecondsSinceEpoch;
 
 TEST(DateArrayAccessor, Test_Date32Array_CDataType_DATE) {
-  std::vector<int32_t> values = {7589, 12320, 18980, 19095};
+  std::vector<int32_t> values = {7589, 12320, 18980, 19095, -1, 0};
+  std::vector<tagDATE_STRUCT> expected = {
+      {1990, 10, 12}, {2003, 9, 25},  {2021, 12, 19},
+      {2022, 4, 13},  {1969, 12, 31}, {1970, 1, 1},
+  };
```

Comment:
Suggestion: could we add a test for a date with year before 1400? E.g., `{1399, 2, 14}`

Reply:
Int_32 minimal -2147483648 -> 1901-12-13 20:45:52.000, but I added 1200 test to another test file